pkg/kanctl/repositoryserver.go

```diff
@@ -0,0 +1,233 @@
+	repoServerUserAccessSecretFlag      = "repositoryServerUserAccessSecret"
+	repoServerAdminUserAccessSecretFlag = "repositoryServerAdminUserAccessSecret"
+	repoPasswordSecretFlag              = "repositoryPasswordSecret"
+	repoUserFlag                        = "repositoryUser"
+	locationCredsSecretFlag             = "locationCredsSecret"
+	locationSecretFlag                  = "locationSecret"
+	defaultKanisterNamespace            = "kanister"
+	defaultRepositoryServerHost         = "localhost"
+)
+
+type repositoryServerParams struct {
+	tls                             string
+	repositoryServerUser            string
+	repositoryServerUserAccess      string
+	repositoryServerAdminUserAccess string
+	repositoryUser                  string
+	repositoryPassword              string
+	prefix                          string
+	location                        string
+	locationCreds                   string
+}
+
+func newRepositoryServerCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "repository-server",
+		Short: "Create a new RepositoryServer",
+		Args:  cobra.ExactArgs(0),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return createNewRepositoryServer(cmd, args)
+		},
+	}
+
+	cmd.PersistentFlags().StringP(tlsSecretFlag, "t", "", "name of the tls secret needed for secure kopia client and kopia repository server communication")
+	cmd.PersistentFlags().StringP(repoServerUserFlag, "u", "", "name of the user to be created for the kopia repository server")
+	cmd.PersistentFlags().StringP(repoServerUserAccessSecretFlag, "k", "", "name of the secret having access credentials of the users that can connect to kopia repository server")
+	cmd.PersistentFlags().StringP(repoServerAdminUserAccessSecretFlag, "a", "", "name of the secret having admin credentials to connect to connect to kopia repository server")
+	cmd.PersistentFlags().StringP(repoPasswordSecretFlag, "r", "", "name of the secret containing password for the kopia repository")
+	cmd.PersistentFlags().StringP(prefixFlag, "p", "", "prefix to be set in kopia repository")
+	cmd.PersistentFlags().StringP(repoUserFlag, "z", "", "name of the user for accessing the kopia repository")
+	cmd.PersistentFlags().StringP(locationSecretFlag, "l", "", "name of the secret containing kopia repository storage location details")
+	cmd.PersistentFlags().StringP(locationCredsSecretFlag, "c", "", "name of the secret containing kopia repository storage credentials")
+
+	_ = cmd.MarkFlagRequired(tlsSecretFlag)
+	_ = cmd.MarkFlagRequired(repoServerUserFlag)
+	_ = cmd.MarkFlagRequired(repoServerUserAccessSecretFlag)
+	_ = cmd.MarkFlagRequired(repoServerAdminUserAccessSecretFlag)
+	_ = cmd.MarkFlagRequired(repoPasswordSecretFlag)
+	_ = cmd.MarkFlagRequired(prefixFlag)
+	_ = cmd.MarkFlagRequired(locationSecretFlag)
+	_ = cmd.MarkFlagRequired(locationCredsSecretFlag)
+	return cmd
+}
+
+func createNewRepositoryServer(cmd *cobra.Command, args []string) error {
+	cmd.SilenceUsage = true
+	if len(args) > 0 {
```

Comment:
what do we get in `args`?

Reply:
it's for positional arguments. We're setting them to zero. And here we're just checking if we got any positional arguments or not